### ISSUE TITLE
passing a timedelta object as payload results in a webui crash

### ISSUE THREAD

**Issue by ZeeD** [mastak/airflow_multi_dagrun#16]
simple example that trigger the webui "crash":

create a pair of DAGS

```python
import datetime

from airflow import DAG
from airflow.operators.dummy_operator import DummyOperator
from airflow.operators.multi_dagrun import TriggerMultiDagRunOperator
from airflow.utils.dates import days_ago

default_args = {
    'owner': '...',
    'depends_on_past': False,
    'start_date': days_ago(1),
    'retries': 0,
    'provide_context': True
}

with DAG('dag1',
         default_args=default_args,
         schedule_interval=datetime.timedelta(days=1)) as dag1:
    op = TriggerMultiDagRunOperator(task_id='op',
                                    trigger_dag_id='dag2',
                                    python_callable=lambda **_: iter([datetime.timedelta(seconds=123.456)]))

with DAG('dag2', default_args=default_args, schedule_interval=None) as dag2:
    dummy_operator = DummyOperator(task_id='dummy_operator')
```

"deploy" them into airflow, and start `dag1` .
After the first run of `dag2`, open its "Tree View".
Instead of the tree view, an error page with a stacktrace ending with `TypeError: Object of type timedelta is not JSON serializable` is shown.

In this regard the behavior of "TriggerDagRunOperator" is different because the payload is `dumps`'d during the creation of the conf attribute:

https://github.com/apache/airflow/blob/c8597cbf143b970ad3c7b0d62e3b44d1dfdc8afe/airflow/operators/dagrun_operator.py#L91-L95

(using airflow-multi-dagrun==1.2 and apache-airflow==1.10.7)

**Comment by mastak** [comment]
Hi @ZeeD 
The same situation with built-in TriggerDagRunOperator if you try to return a non-serializable object from python_callable. That why it's not an issue with the current plugin. This is the limitation of the airflow.

**Comment by ZeeD** [comment]
Hi.
I can understand you reasoning behind the "`dumps`able" requirement of the object.
Unfortunately the error happeded in a bit of "wrong" time (after the deploy and some runs), so I think a pair of actions could be done:

1) document the limitation: maybe I couldn't find it, but it seems to me there is no mention of this requirement, and I knew about it only after looking inside the source code.
2) if it is possible, "fail fast": maybe you could serialize the data as soon as possible, as the "TridderDagRunOperator" does. In this way the limitation is clear.
3) (maybe this is more a "feature request", but I'll add it) Add an optional parameter "serializer" to let the user inject a replacer for `dumps` (or the `cls` parameter of it)

**Comment by eff917** [comment]
I know it's closed, but airflow 2.3.0 grid view (the successor of tree view) now works.
